### PR TITLE
pagination-added-to-tests

### PR DESCRIPTION
**Pull request by yash6102000**
pagination added to tests
where data is fetching according to page size and current page number

### CODE REVIEW

File: app/components/sections/Sections.tsx [hunk 1]
```diff
@@ -28,12 +32,17 @@ const SectionLink = ({
...
+  testCurrentPage: number
+  testCurrentItems: number
+  totalCount: number
 }) => {
   const path = `/${currentWorkspaceId}${routes.tests}/${section.id}${filter}`
   const [isDelete, setIsDelete] = useState(false)
   const location = useLocation() // to get current location
   const resolvedPath = useResolvedPath(path) // to get resolved path which would match with current location
   const isActive = location.pathname === resolvedPath.pathname
+  console.log(isActive, 'isActive')
```
Comment (richa-copods) [inline app/components/sections/Sections.tsx:44]: Remove consoles

File: app/components/sections/Sections.tsx [hunk 2]
```diff
@@ -122,7 +134,19 @@ const Sections = ({
   currentWorkspaceId,
   sectionActionErrors,
   setSectionActionErrors,
+  testCurrentPage,
+  testCurrentItems,
+  totalCount,
 }: SectionType) => {
+  let navigate = useNavigate()
```
Comment (richa-copods) [inline app/components/sections/Sections.tsx:141]: Use `const` here

File: app/components/sections/Sections.tsx [hunk 3]
```diff
@@ -122,7 +134,19 @@ const Sections = ({
...
+  const [testsPageSize, setTestPageSize] = useState(5)
+  const [testsCurrentPage, setTestsCurrentPage] = useState(testCurrentPage)
+  useEffect(() => {
+    console.log('navigate')
```
Comment (richa-copods) [inline app/components/sections/Sections.tsx:145]: Remove console

File: app/models/sections.server.ts [hunk 4]
```diff
@@ -38,11 +38,29 @@ export async function getFirstSection(workspaceId: string) {
 
   return firstSection?.id
 }
-
-export async function getAllSections(filterData: string, workspaceId: string) {
+export async function getAllTestsCounts(
+  currentWorkspaceId: string | undefined
```
Comment (richa-copods) [inline app/models/sections.server.ts:42]: Can `currentWorkspaceId` be ever undefined?

File: app/routes/$workspaceId/tests.tsx [hunk 5]
```diff
@@ -151,25 +155,36 @@ const handleEditSection = async (
   return editHandle
 }
 export const loader: LoaderFunction = async ({ request, params }) => {
+  const query = new URL(request.url).searchParams
+  const testItemsPerPage = Math.max(Number(query.get('testItems') || 5), 5) //To set the lower bound, so that minimum count will always be 1 for current page and 5 for items per page.
+  const testCurrentPage = Math.max(Number(query.get('testPage') || 1), 1)
+
   const userId = await getUserId(request)
   const currentWorkspaceId = params.workspaceId as string
+  console.log(testCurrentPage, 'testCurrentPage')
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests.tsx:164]: Remove the console

File: app/routes/$workspaceId/tests.tsx [hunk 6]
```diff
@@ -297,7 +315,7 @@ export const action: ActionFunction = async ({ request, params }) => {
 
 export default function SectionPage() {
   const data = useLoaderData() as unknown as LoaderData
-
+  console.log(data, 'data')
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests.tsx:318]: Remove all the consoles

File: app/routes/$workspaceId/tests.tsx [hunk 7]
```diff
@@ -321,7 +339,7 @@ export default function SectionPage() {
     description: '',
   })
   const [selectedSection, setSelectedSection] = useState(
-    data.selectedSectionId || data.sections[0]?.id || 'NA'
+    data.selectedSectionId || data.sections[6]?.id || 'NA'
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests.tsx:342]: Why using 6th index?

File: app/routes/$workspaceId/tests.tsx [hunk 8]
```diff
@@ -391,7 +409,7 @@ export default function SectionPage() {
     }
   }, [
     sectionActionData,
-    data.selectedSectionId,
+    // data.selectedSectionId,
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests.tsx:412]: Remove unwanted commented code

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/sections/Sections.tsx (hunk 2 above)
Comment (richa-copods) [reply app/components/sections/Sections.tsx:139]: Resolve this

Comment (richa-copods) [reply app/components/sections/Sections.tsx:139]: Resolve this

File: app/models/sections.server.ts [hunk 9]
```diff
@@ -38,11 +38,27 @@ export async function getFirstSection(workspaceId: string) {
 
   return firstSection?.id
 }
-
-export async function getAllSections(filterData: string, workspaceId: string) {
+export async function getAllTestsCounts(currentWorkspaceId?: string) {
```
Comment (richa-copods) [inline app/models/sections.server.ts:41]: Shouldn't passing `currentWorkspaceId` be mandatory?

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/models/sections.server.ts (hunk 9 above)
Comment (yash6102000) [reply app/models/sections.server.ts:41]: cuz we are fetching as per active workspace/current workspace

Comment (anuragP04) [reply app/models/sections.server.ts:41]: Add filter parameter in count.

Comment (yash6102000) [reply app/models/sections.server.ts]: filter isn't needed

File: app/components/sections/Sections.tsx [hunk 10]
```diff
@@ -122,7 +132,18 @@ const Sections = ({
...
+  testCurrentPage,
+  testCurrentItems,
+  totalCount,
 }: SectionType) => {
+  const navigate = useNavigate()
+  const [testsPageSize, setTestPageSize] = useState(5)
+  const [testsCurrentPage, setTestsCurrentPage] = useState(testCurrentPage)
+  useEffect(() => {
+    navigate(
+      `?filter=%7B"orderBy"%3A%7B"createdAt"%3A"desc"%7D%7D&testPage=${testsCurrentPage}&testItems=${testsPageSize}`
```
Comment (richa-copods) [inline app/components/sections/Sections.tsx:144]: Simplify the query parameters string

File: app/routes/$workspaceId/tests.tsx [hunk 11]
```diff
@@ -342,7 +358,7 @@ export default function SectionPage() {
       formData.append('filter', JSON.stringify(filter))
       submit(formData, {
         method: 'get',
-        action: `/${data.currentWorkspaceId}${routes.tests}/${selectedSection}`,
+        action: `/${data.currentWorkspaceId}${routes.tests}/${data.sections[0].id}?filter=%7B"orderBy"%3A%7B"createdAt"%3A"desc"%7D%7D&testPage=${data.testCurrentPage}&testItems=${data.testItemsPerPage}`,
```
Comment (richa-copods) [inline app/routes/$workspaceId/tests.tsx:361]: Simplify the query parameters string

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: app/components/sections/Sections.tsx [hunk 12]
```diff
@@ -6,6 +6,7 @@ import SortFilter from '../common-components/SortFilter'
 import { useEffect, useState } from 'react'
 import { routes } from '~/constants/route.constants'
 import type { sectionActionErrorsType } from '~/interface/Interface'
+import Pagination from '../common-components/Pagination'
```
Comment (anuragP04) [inline app/components/sections/Sections.tsx:9]: Breaking UI

File: app/components/sections/Sections.tsx [hunk 13]
```diff
@@ -6,6 +6,7 @@ import SortFilter from '../common-components/SortFilter'
...
```
Comment (anuragP04) [inline app/components/sections/Sections.tsx:10]: Breaking UI

File: app/components/sections/Sections.tsx [hunk 14]
```diff
@@ -6,6 +6,7 @@ import SortFilter from '../common-components/SortFilter'
...
 
 const SectionLink = ({
   section,
```
Comment (anuragP04) [inline app/components/sections/Sections.tsx:12]: delete issue is still not fixed, if i'm on a page with single record and delete it, I am stuck with `no record found`

File: app/routes/$workspaceId/tests.tsx [hunk 15]
```diff
@@ -361,14 +374,56 @@ export default function SectionPage() {
...
+    testsCurrentPage,
+    testsPageSize,
+    sortBy,
+    order,
+    data.lastSectionId?.id,
     t,
-    location,
-    navigate,
+    data.getAllTestsCount,
+    data.sections[0]?.id,
```
Comment (anuragP04) [inline app/routes/$workspaceId/tests.tsx:423]: why we have added this dependency here?

File: app/routes/$workspaceId/assessments.add-assessment.tsx [hunk 16]
```diff
@@ -34,18 +34,11 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
-      )
-    : '{}'
-
   let sections: Array<Section> = []
   let status: string = ''
-  await getAllSections(filter, currentWorkspaceId as string)
+  await getAllSections('', '', currentWorkspaceId as string)
     .then((res) => {
-      sections = res as Section[]
+      sections = res as unknown as Section[]
```
Comment (anuragP04) [inline app/routes/$workspaceId/assessments.add-assessment.tsx:41]: why unknown??

File: app/routes/$workspaceId/assessments.add-assessment.tsx [hunk 17]
```diff
@@ -34,18 +34,11 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
-    .data
-    ? JSON.parse(
-        Object.fromEntries(new URL(request.url).searchParams.entries()).data
-      )
-    : '{}'
-
   let sections: Array<Section> = []
   let status: string = ''
-  await getAllSections(filter, currentWorkspaceId as string)
+  await getAllSections('', '', currentWorkspaceId as string)
```
Comment (anuragP04) [inline app/routes/$workspaceId/assessments.add-assessment.tsx:39]: why we are not sending filters here?

File: app/components/sections/tests.helper.tsx [hunk 18]
```diff
@@ -0,0 +1,25 @@
+import { getAllSections } from '~/models/sections.server'
```
Comment (anuragP04) [inline app/components/sections/tests.helper.tsx:1]: discuss with me regarding this file

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**